pysoundcomparisons/mediacatalog.py

```diff
@@ -1,31 +1,26 @@
 from cdstarcat import Catalog
+from clldutils.misc import lazyproperty
 
 class MediaCatalog(Catalog):
-    __mimetypes = {
+
+    valid_mimetypes = {
         'mp3': 'audio/mpeg',
         'ogg': 'audio/ogg',
         'wav': 'audio/wav',
     }
 
-    __file_path_uid_map = None
-
+    @lazyproperty
     def file_path_uid_map(self):
-        if self.__file_path_uid_map is None:
-            self.__file_path_uid_map = { 
-                self.objects[i].metadata['name'] : i for i in self.objects}
-        return self.__file_path_uid_map
+        return {obj.metadata['name'] : obj.id for obj in self}
```

Comment:
Judging by the way `file_path_uid_map` is used, I'd say it should rather be replaced by a custom `__getitem__` implementation:
```python
    def __getitem__(self, key):
        return Catalog.__getitem__(self, key) or self.file_path_uid_map.get(key)
```

Reply:
I.e. `file_path_uid_map` should return a `dict` with objects as values, not UIDs. I've seen at least two usages, where the UID was retrieved to then lookup the object, e.g.
```python
            if sfpath in catalog.file_path_uid_map:
                uid = catalog.file_path_uid_map[sfpath]
                bs_obj = catalog.objects[uid].bitstreams
```
with the above changes, this would be:
```python
    bs_obj = catalog[sfpath].bitstreams
```

Reply:
`__contains__` should be customized, too, then we could treat file paths as alternative catalog keys.

Reply:
ok - but then we have to change

    def file_path_uid_map(self):
        return {obj.metadata['name'] : obj.id for obj in self}

since this approach, which would save memory, maps `file_path_name` and `uid` as string and not the entire object and 

        bs_obj = catalog.objects[uid].bitstreams

would raise an exception à la `'str' object has no attribute 'bitstreams'`